roles/reproducer/tasks/main.yml

```diff
@@ -142,6 +142,33 @@
       }}
   failed_when: false
 
+- name: Ensure custom networks can be routed outside to the hypervisor
+  when:
+    - cifmw_reproducer_nat_public_interfaces | default ('false') | bool
+  become: true
+  block:
+    - name: Enable masquered over public interfaces
+      ansible.posix.firewalld:
+        masquerade: true
+        zone: public
+        permanent: true
+        immediate: true
+        state: enabled
+
+    - name: Ensure network is in libvirt zone
```

Comment:
I think it would be better to change the `ci_nmstate`role - add a `zone` parameter where interfaces are defined there. So you set:

```
- name: Configure a couple of nets using ci_nmstate in two Ansible hosts
      vars:
        cifmw_ci_nmstate_network_layout:
            instance-1:
              default:
                connection: ci-private-network
                zone: _your_zone_of_choice_
```

You can even add that in the more advanced "direct" thing - cifmw_ci_nmstate_instance_config - it should be trivial to filter out the zone before passing the data to nmstate.